docs/guides/create-composable-components.stories.mdx

```diff
@@ -0,0 +1,187 @@
+   [Combobox](..?path=/docs/components-selections-combobox--docs)
+
+   These often have strict APIs, with customization limited to specific props.
+   Internal UI and behaviors are not easily modifiable.
+
+## Path Forward: Favoring a Custom Render Function
+
+To address these challenges, we are adopting a **custom render function**
+approach as a preferred pattern for composability.
+
+#### Named Render Prop: `customRender____`
+
+This works by providing a named render prop (e.g., `customRenderItem`) that
+allows consumers to inject their own UI while preserving default behaviors. For
+example:
+
+```tsx
+const renderProductItem = item => (
+  <Flex template={["shrink", "grow"]} align="start">
+    <Text>{item.price}</Text>
+    <Heading level={4}>{item.name}</Heading>
+  </Flex>
+);
+
+export const CustomRenderExample = () => (
+  <List items={items} customRenderItem={renderProductItem} />
+);
+```
+
+**Why This Approach?**
+
+- **Clarity**: The `customRender` prefix clearly communicates the desire to do
+  advanced customization.
+- **Flexibility**: Supports both simple and complex components without major
+  refactoring.
+- **Trackability**: Easy to search for and monitor across codebases.
+
+### When a Custom Render Function Might Not Be the Right Solution
+
+While custom render functions are powerful, there are cases where they may not
+be the optimal choice.
+
+#### Example: `Tab` Component (see [Tabs](..?path=/docs/components-navigation-tabs--docs))
+
+Instead of introducing `customRenderLabel`, the `Tab` component was updated to
+allow the `label` prop to accept a `ReactNode`.
+
+```tsx
+<Tab label={<MyCustomLabel />} />
+```
+
+This approach was chosen for the following reasons:
+
+1. **Incremental Improvement**: Extending the `label` prop type was a simpler
+   change that still met customization needs.
+2. **No Internal State Exposure**: Customizing the `label` didn’t require
+   exposing or interacting with the `Tab`’s internal state (e.g., active state
+   styling).
+3. **Clarity of Purpose**: The `children` prop is already used for other
+   purposes, and adding a render function could create unnecessary complexity.
+
+### Considerations When Deciding on a Custom Render Function
+
+When deciding whether to use a custom render function, consider the following
+factors:
+
+1. **Internal State Requirements**:
+
+   - If customization depends on internal state (e.g., active status), a custom
+     render function might be necessary.
+
+2. **Atomic vs. Compound Components**:
+
+   - For atomic components like `Tab`, small changes like accepting `ReactNode`
+     might suffice.
+   - For compound components with multiple customizable parts, render functions
+     provide clearer APIs.
+
+3. **Existing API and Props**:
+
+   - Avoid overloading components with too many customization methods. Leverage
+     existing props like `children` where appropriate.
+
+4. **Context and Identity**:
```

Comment:
this is definitely important, but I'd say it's most important for us (Atlantis) when reviewing, or thinking about what we want to allow to be customizable. traditionally I don't think of the guides on Atlantis as being for us, but maybe that's not the right way to think about that.

I'm trying to decide if I think that's beneficial for others to know what we are going to use as criteria?